### PR TITLE
Add logic to dynamically discover primary interface name

### PR DESCRIPTION
**Pull request by liwenwu-amazon**
*Issue #171 , if available:*

*Description of changes:*
* use metadata service to find out the mac address on primary interface
* go through all the interfaces and find the interface which has same mac address as primary interface's mac.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/awsutils/awsutils.go [hunk 1]
```diff
@@ -124,6 +124,8 @@ type APIs interface {
 
 	// GetENILimit returns the number of enis can be attached to an instance
 	GetENILimit() (int, error)
+	// GetPrimaryENIMAC returns the mac address of primary eni
```
Comment (mogren) [inline pkg/awsutils/awsutils.go]: Nit: `// GetPrimaryENIMAC returns the mac address of the primary ENI`

File: pkg/networkutils/network.go [hunk 2]
```diff
@@ -163,9 +194,11 @@ func (n *linuxNetwork) SetupHostNetwork(vpcCIDR *net.IPNet, primaryAddr *net.IP)
 		// - Thus, it finds the source-based route that leaves via the secondary ENI.
 		// - In "strict" mode, the RPF check fails because the return path uses a different interface to the incoming
 		//   packet.  In "loose" mode, the check passes because some route was found.
-		const eth0RPFilter = "/proc/sys/net/ipv4/conf/eth0/rp_filter"
+		eth0RPFilter := "/proc/sys/net/ipv4/conf/" + primaryIntf + "/rp_filter"
```
Comment (mogren) [inline pkg/networkutils/network.go]: `eth0RPFilter` -> `primaryIntfRPFilter`

File: pkg/networkutils/network.go [hunk 3]
```diff
@@ -126,9 +126,34 @@ func isDuplicateRuleAdd(err error) bool {
 	return strings.Contains(err.Error(), "File exists")
 }
 
+// find out the primary interface name
+func findPrimaryInterfaceName(primaryMAC string) (string, error) {
+
+	log.Debugf("Trying to find primary interface that has mac : %s", primaryMAC)
+
+	interfaces, err := net.Interfaces()
```
Comment (mogren) [inline pkg/networkutils/network.go]: This call to list all interfaces is probably the biggest change. 😄

**Comment by liwenwu-amazon** [comment]
@ewbankkit @phyllisstein,  can you verify if this PR fixed your CNI in your environment?  thanks

**Comment by phyllisstein** [comment]
@liwenwu-amazon This is awesome, thanks so much for taking up the issue! I just migrated from Kops with Ubuntu back to EKS, but I'll try rebuilding my Kops cluster again with this change in the next couple of days.

**Comment by ewbankkit** [comment]
Yes, I should be able to verify in the next couple of days too.

File: pkg/awsutils/awsutils.go (hunk 1 above)
Comment (liwenwu-amazon) [reply pkg/awsutils/awsutils.go]: done

File: pkg/networkutils/network.go (hunk 3 above)
Comment (liwenwu-amazon) [reply pkg/networkutils/network.go]: @mogren I just unit tested the `net.Interfaces()` call on ubuntu t3.large instance. It is working as expected.
Here is the test program
```
cat test.go
package main

import (
	"fmt"
	"net"
)

func main() {

	interfaces, _ := net.Interfaces()

	for _, intf := range interfaces {
		fmt.Printf("interface %v, mac %v\n", intf.Name, intf.HardwareAddr)
	}
}
```
Here is the test output
```
root@ip-172-31-0-186:/home/ubuntu# ./test
interface lo, mac 
interface ens5, mac 0a:5f:92:e3:59:86
```
Here is the output of ifconfig
```
root@ip-172-31-0-186:/home/ubuntu# ifconfig
ens5: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 9001
        inet 172.31.0.186  netmask 255.255.240.0  broadcast 172.31.15.255
        inet6 fe80::85f:92ff:fee3:5986  prefixlen 64  scopeid 0x20<link>
        ether 0a:5f:92:e3:59:86  txqueuelen 1000  (Ethernet)
        RX packets 6072  bytes 7297970 (7.2 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 1699  bytes 199849 (199.8 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 191  bytes 14956 (14.9 KB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 191  bytes 14956 (14.9 KB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

**Comment by ewbankkit** [comment]
@liwenwu-amazon I have verified that building from the latest `master` source and running the CNI plugin on `CentOS Linux release 7.5.1804` is now successful.

**Comment by phyllisstein** [comment]
Confirmed that v1.2.1 does the trick with Ubuntu 18.04 as well. Thanks again!

**Pull request merged**